addon/components/frost-modal-input.js

```diff
@@ -1,9 +1,140 @@
+  scrollBindingsSet: false,
+  containerObserver: null,  // @type {window.MutationObserver}
+  $containerEl: null,
+  $headerEl: null,
+  $footerEl: null,
+
+  /**
+   * create a window.MutationObserver on the container element
+   * mutation events callback invokes updateScrollStyles()
+   * @returns {MutationObserver} window.MutationObserver on container element
+   */
+  createMutationObserver () {
+    // create an observer instance
+    let mutationObserver = new window.MutationObserver((mutations) => {
+      mutations.forEach((mutation) => {
+        this.updateScrollStyles()
+      })
+    })
+
+    // configuration of the observer
+    const config = { attributes: false, childList: true, characterData: false, subtree: true }
+    // pass in the target DOM node, as well as the observer options
+    mutationObserver.observe(this.get('$containerEl').get(0), config)
+
+    return mutationObserver
+  },
+
+  /**
+  * update perfect-scrollbar before checking scoll position
+  * need to disconnect MutationObserver then re-establish to prevent infinite loop
+  * of mutation event triggers, callback updates, which triggers new mutation event...
+  */
+  updatePerfectScrollbar () {
+    let mutationObserver
+    const $containerEl = this.get('$containerEl')
+
+    Ps.update($containerEl.get(0))
+
+    if (this.get('containerObserver')) {
+      this.get('containerObserver').disconnect()
+      mutationObserver = this.createMutationObserver()
+    }
+
+    if (this.isDestroyed || this.isDestroying) {
+      mutationObserver.disconnect()
+      return
+    }
+
+    this.set('containerObserver', mutationObserver)
+  },
+
+  /**
+  * Checks scroll position within container element to add/remove scroll styling to header/footer elements
+  */
+  updateScrollStyles () {
+    const $containerEl = this.get('$containerEl')
+    const $headerEl = this.get('$headerEl')
+    const $footerEl = this.get('$footerEl')
+    const scrollTop = $containerEl.scrollTop()
+    const innerHeight = $containerEl.innerHeight()
+    const scrollHeight = $containerEl.get(0) && $containerEl.get(0).scrollHeight
+    this.updatePerfectScrollbar()
+
+    // style top of form with box shadow if content overflow hidden underneath header
+    const shouldHeaderClassBePresent = scrollTop > 0
+    $headerEl.toggleClass('header-scrolled', shouldHeaderClassBePresent)
+
+    // style bottom of form with box shadow if content overflow hidden underneath footer
+    const shouldFooterClassBePresent = scrollTop + innerHeight < scrollHeight
+    $footerEl.toggleClass('footer-scrolled', shouldFooterClassBePresent)
+  },
+
+  /**
+   * sets bindings/callbacks for perfect-scrollbar events on document element
+   * creates a window.MutationObserver for mutating events on container element
+   */
+  setScrollBindings () {
+    const $headerEl = this.get('$headerEl')
+    const $footerEl = this.get('$footerEl')
+    // const $containerEl = this.get('$containerEl')
```

Comment:
Might want to remove these commented out code sections in this method. I see a few more below.